src/full/Agda/TypeChecking/Errors.hs

```diff
@@ -1248,6 +1249,35 @@ instance PrettyTCM TypeError where
     MultiplePolarityPragmas xs -> fsep $
       pwords "Multiple polarity pragmas for" ++ map pretty (List1.toList xs)
 
+    ConstructorNameOfNonRecord res -> case res of
+      UnknownName -> __IMPOSSIBLE__ -- Turned into NotInScope when the name is resolved
+      name ->
+        let
+          qn :: m Doc
+          (qn, whatis) = case name of
+            DefinedName _ nm _ -> (prettyTCM nm,) case anameKind nm of
+              RecName                  -> __IMPOSSIBLE__
+
+              ConName                  -> "a constructor"
+              CoConName                -> "a constructor"
+              FldName                  -> "a record field"
+              PatternSynName           -> "a pattern synonym"
```

Comment:
Are all of these possible? 
https://github.com/agda/agda/blob/19c46ee1b13bf6a576ae0be7d450f4355f70e5e2/src/full/Agda/Syntax/Scope/Base.hs#L520-L527
If yes the comment on `DefinedName` in the linked source is outdated.
If no, then please use `__IMPOSSIBLE__`.